common/src/main/scala/com/gu/media/Permissions.scala

```diff
@@ -41,3 +51,11 @@ class MediaAtomMakerPermissionsProvider(stage: String, credsProvider: AWSCredent
     }
   }
 }
+
+object PermissionsUploadHelper {
+  def permissionAndIntentMatch(permission: Permissions, selfHosted: Boolean) : Boolean = {
```

Comment:
I had it in the controller but then wanted to put a test around so that we catch ourselves if we make changes to it later. So pulled it out to unit test it easier. Happy to rename it.